Problem with run check.py on MacOS

It seems script assume that all files in `/binaryen/bin` are executable but in some systems like MacOS filesystem can insert hidden aux files and caches like `.DS_Store`.

```bash
./check.py
[ checking --version ... ]

Parsed CHANGELOG.md version: 101

.. <path>/binaryen/bin/.DS_Store --version
Traceback (most recent call last):
  File " <path>/binaryen/./check.py", line 393, in <module>
    sys.exit(main())
  File " <path>/binaryen/./check.py", line 376, in main
    TEST_SUITES[test]()
  File " <path>binaryen/./check.py", line 54, in run_version_tests
    out, err = subprocess.Popen([e, '--version'],
  File " <path>/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File " <path>/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 13] Permission denied: ' <path>/binaryen/bin/.DS_Store'
```

cc @tlively

Hmm, after fixed this issue I got another one:
```
Parsed CHANGELOG.md version: 101

.. /Volumes/Archive/Projects/Github/binaryen/bin/asm2wasm --version
Traceback (most recent call last):
  File "<path>/binaryen/./check.py", line 393, in <module>
    sys.exit(main())
  File "<path>/binaryen/./check.py", line 376, in main
    TEST_SUITES[test]()
  File "<path>/binaryen/./check.py", line 59, in run_version_tests
    assert len(err) == 0, 'Expected no stderr, got:\n%s' % err
AssertionError: Expected no stderr, got:
dyld: Symbol not found: __ZN4wasm11GLOBAL_MATHE
  Referenced from: <path>/Github/binaryen/bin/asm2wasm
  Expected in:<path>/binaryen/bin/../lib/libbinaryen.dylib
 in <path>/binaryen/bin/asm2wasm
```

--- a/check.py
+++ b/check.py
@@ -42,7 +42,7 @@ def get_changelog_version():
 def run_version_tests():
     print('[ checking --version ... ]\n')
 
-    not_executable_suffix = ['.txt', '.js', '.ilk', '.pdb', '.dll', '.wasm', '.manifest', 'binaryen-lit']
+    not_executable_suffix = ['.DS_Store', '.txt', '.js', '.ilk', '.pdb', '.dll', '.wasm', '.manifest', 'binaryen-lit']
     bin_files = [os.path.join(shared.options.binaryen_bin, f) for f in os.listdir(shared.options.binaryen_bin)]
     executables = [f for f in bin_files if os.path.isfile(f) and not any(f.endswith(s) for s in not_executable_suffix)]
     executables = sorted(executables)

